### PR TITLE
Upgrade scikit-image to 0.24.0

### PR DESCRIPTION
**Pull request by mkcor**
Dear Pyodide community,

We are teaching a [scikit-image tutorial](https://pretalx.com/euroscipy-2024/talk/ZVBAKK/) at EuroSciPy 2024 in just a couple of weeks. Following in @glemaitre's [footsteps](https://github.com/glemaitre/euroscipy-2023-scikit-image), we thought we would use JupyterLite and rely on the Pyodide distribution.

We'll be fine running scikit-image 0.23.2 but *ideally* we would like to run scikit-image 0.24.0 (and NumPy version 2)! Of course, we don't *expect* that you will release a new version by August 26th, but we might as well try and submit this PR. I'll add a CHANGELOG entry right away! :innocent: 

/cc @lagru @stefanv

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

File: docs/project/changelog.md
Comment (agriyakhetarpal) [inline docs/project/changelog.md:66]: Could you please move this up so that it is below one of the "Upgraded" entries and above the "Added" ones?

**Review by agriyakhetarpal — APPROVED** [review]
Thanks for the update, @mkcor! Would you like to add yourself as a maintainer for the recipe in this PR?

File: docs/project/changelog.md
Comment (mkcor) [reply docs/project/changelog.md]: Sure! Done 49b748cb8e7ee7a264f11.

**Comment by mkcor** [comment]
> Thanks for the update, @mkcor! Would you like to add yourself as a maintainer for the recipe in this PR?

Would that simply involve that I can (and/or I'm supposed to?) submit similar PRs each time we release a new minor version?

> Edit: also, NumPy v2 is very much in progress: #4925

Wonderful!

**Comment by agriyakhetarpal** [comment]
> Would that simply involve that I can (and/or I'm supposed to?) submit similar PRs each time we release a new minor version?

Yes, if you'd like to do it and/or include it in the release procedure for `scikit-image`, that would be great! It's not strictly needed, since we run `make -C packages update-all` to bump to the latest versions for all packages every time we make a release – which is also where we bump the CPython version. More frequent bumps through such PRs would mean that `scikit-image` can potentially receive updates with patch releases as well, say, when we release 0.XY.A/0.XY.B/and so on. You could consider automating this with @pyodide-pr-bot, though I notice that it is just `awkward-cpp` that is using it, so far.

Also, having a recipe maintainer would mean that the Pyodide developers and users of `scikit-image` within Pyodide will know who to ping in case there are any problems with using or updating the recipe. :)

**Comment by agriyakhetarpal** [comment]
If it's a yes, please feel free to add more than one maintainer if you would like to!

**Comment by mkcor** [comment]
Got it, thanks, @agriyakhetarpal!

@lagru would you like to co-maintain this recipe? :wink:

**Comment by lagru** [comment]
Yes, I'm willing to give it a try. :wink:

**Comment by mkcor** [comment]
Great, @agriyakhetarpal so that's a yes from both of us!

**Comment by hoodmane** [comment]
@mkcor could you add the two of you as maintainers? See here for what it should look lilke?
https://github.com/pyodide/pyodide/blob/main/packages/scikit-learn/meta.yaml?plain=1#L69

After that, this looks good to merge.

**Comment by hoodmane** [comment]
Thanks @mkcor and @lagru!

**Comment by agriyakhetarpal** [comment]
I think @hoodmane would know this better than me, so I'll let him answer that question, but as for your comment in the PR description:

> We'll be fine running scikit-image 0.23.2 but ideally we would like to run scikit-image 0.24.0 (and NumPy version 2)! Of course, we don't expect that you will release a new version by August 26th

You could build your own Pyodide distribution and use a static site with a JupyterLite deployment with it, containing updated versions of the packages you need. See the docs here: https://jupyterlite.readthedocs.io/en/stable/howto/pyodide/pyodide.html

I hope your tutorial at EuroSciPy 2024 goes well!

**Comment by hoodmane** [comment]
I can release 0.27.0a1 if you're willing to use an alpha release for your class. Looking at the changelog there is nothing particularly scary in there so it should work fine.

**Comment by lagru** [comment]
Following [custom Pyodide distribution](https://jupyterlite.readthedocs.io/en/latest/howto/pyodide/pyodide.html) I managed to find an [artifact on CircelCI](https://app.circleci.com/pipelines/github/pyodide/pyodide/7707/workflows/0dc18659-38ad-41e1-af34-ae54fcb82b57/jobs/110674/artifacts) that apparently contains all libraries. Extracting and building it with jupyterlite seemed to go fine but I assume that I missed something. The kernel seems to load forever even with a simple command like `1 + 1`. I think I setup libarchive-c correctly.

Would an alpha release help with that process? I assume that following an alpha release I'd be able to point  `pyodideUrl` from `jupyter_lite.json` to `https://cdn.jsdelivr.net/pyodide/v0.27.0a1/full/pyodide.js` and it would "just work", correct?

**Comment by ryanking13** [comment]
@lagru I don't know about how jupyterlite works with custom Pyodide distribution, but yes, I think releasing an alpha version would help.

**Comment by hoodmane** [comment]
@lagru @stefanv we released Pyodide 0.27.0a2 so you can potentially use that for your tutorial.

**Comment by glemaitre** [comment]
@lagru @mkcor @stefanv In case this is useful for your tutorial, we set up our jupyterlite for this year that uses the pyodide dev version: https://github.com/glemaitre/calibration-cost-sensitive-learning-jupyterlite

The only required change is this file: https://github.com/glemaitre/calibration-cost-sensitive-learning-jupyterlite/blob/main/jupyter-lite.json

This is inline with what you @lagru already mentioned and shows that it works as expected.

**Comment by hoodmane** [comment]
I'd recommend that you pin a version for a class since we deploy the dev version every time we commit to main and the deploy usually works but sometimes dev is broken for a few hours.

**Comment by lagru** [comment]
[We got it working now](https://github.com/pyodide/pyodide). Thank you all very much for the release and the guidance! :blush: Hope our request didn't cause to much disruption!

**Pull request merged**
